visualization/include/pcl/visualization/impl/pcl_visualizer.hpp

```diff
@@ -669,7 +669,8 @@ pcl::visualization::PCLVisualizer::addText3D (
   else
     tid = id;
 
-  if (contains (tid))
+  std::string idToCheck = viewport > 0 ? tid.append(viewport, '*') : tid;
+  if (contains (idToCheck))
   {
     PCL_WARN ("[addText3D] The id <%s> already exists! Please choose a different id and retry.\n", id.c_str ());
```

Comment:
Perhaps we can improve the warning by additionally specifying in which viewport the id already exists?

Reply:
good idea, will add.